crates/storage/provider/src/bundle_state/state_reverts.rs

```diff
@@ -1,108 +1,33 @@
-            {
-                let storage_id = BlockNumberAddress((block_number, address));
-
-                let mut storage = storage_revert
-                    .into_iter()
-                    .map(|(k, v)| (B256::new(k.to_be_bytes()), v))
-                    .collect::<Vec<_>>();
-                // sort storage slots by key.
-                storage.par_sort_unstable_by_key(|a| a.0);
-
-                // If we are writing the primary storage wipe transition, the pre-existing plain
-                // storage state has to be taken from the database and written to storage history.
-                // See [StorageWipe::Primary] for more details.
-                let mut wiped_storage = Vec::new();
-                if wiped {
-                    tracing::trace!(target: "provider::reverts", ?address, "Wiping storage");
-                    if let Some((_, entry)) = storages_cursor.seek_exact(address)? {
-                        wiped_storage.push((entry.key, entry.value));
-                        while let Some(entry) = storages_cursor.next_dup_val()? {
-                            wiped_storage.push((entry.key, entry.value))
-                        }
-                    }
-                }
-
-                tracing::trace!(target: "provider::reverts", ?address, ?storage, "Writing storage reverts");
-                for (key, value) in StorageRevertsIter::new(storage, wiped_storage) {
-                    storage_changeset_cursor.append_dup(storage_id, StorageEntry { key, value })?;
-                }
-            }
-        }
-
-        // Write account changes
-        tracing::trace!(target: "provider::reverts", "Writing account changes");
-        let mut account_changeset_cursor = tx.cursor_dup_write::<tables::AccountChangeSet>()?;
-        for (block_index, mut account_block_reverts) in self.0.accounts.into_iter().enumerate() {
-            let block_number = first_block + block_index as BlockNumber;
-            // Sort accounts by address.
-            account_block_reverts.par_sort_by_key(|a| a.0);
-            for (address, info) in account_block_reverts {
-                account_changeset_cursor.append_dup(
-                    block_number,
-                    AccountBeforeTx { address, info: info.map(into_reth_acc) },
-                )?;
-            }
-        }
-
-        Ok(())
-    }
-}
-
 /// Iterator over storage reverts.
 /// See [StorageRevertsIter::next] for more details.
-struct StorageRevertsIter<R: Iterator, W: Iterator> {
+pub struct StorageRevertsIter<R: Iterator, W: Iterator> {
     reverts: Peekable<R>,
     wiped: Peekable<W>,
 }
 
+impl<R, W, A, B> std::fmt::Debug for StorageRevertsIter<R, W>
+where
+    R: Iterator<Item = A>,
+    W: Iterator<Item = B>,
+    A: std::fmt::Debug,
+    B: std::fmt::Debug,
+{
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        f.debug_struct("StorageRevertsIter").finish()
+    }
+}
```

Comment:
isn't this the same as deriving debug but now without including the conent

Reply:
yeah, can't derive debug though, mainly did this for the missing debug impl lint on public items. Does debug make sense here? I can add contents if so